### DISCUSSION TITLE
Error when trying to run stable diffusion for Intel arc using Jupyter notebooks. Kubuntu 22.04

### DISCUSSION THREAD

**Discussion by nom0to** [openvinotoolkit/openvino_notebooks#1387] [Q&A]
Hello everyone, could really use some help here. I've been trying to run the jupyter notebooks in order to use stable diffusion on my computer but the Jupyter notebooks is either very broken or I'm missing something. I'm currently receiving this error on the Configure Inference Pipeline step: `---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget_selection.py:133, in findvalue(array, value, compare)
    132 try:
--> 133     return next(x for x in array if compare(x, value))
    134 except StopIteration:

StopIteration: 

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget_selection.py:242, in _Selection._validate_value(self, proposal)
    241 try:
--> 242     return findvalue(self._options_values, value, self.equals) if value is not None else None
    243 except ValueError:

File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget_selection.py:135, in findvalue(array, value, compare)
    134 except StopIteration:
--> 135     raise ValueError('%r not in array'%value)

ValueError: 'GPU' not in array

During handling of the above exception, another exception occurred:

TraitError                                Traceback (most recent call last)
Cell In[7], line 3
      1 import ipywidgets as widgets
----> 3 device = widgets.Dropdown(
      4     options=core.available_devices + ["AUTO"],
      5     value='GPU',
      6     description='Device:',
      7     disabled=False,
      8 )
     10 device

File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget_selection.py:186, in _Selection.__init__(self, *args, **kwargs)
    183     kwargs['index'] = 0 if nonempty else None
    184     kwargs['label'], kwargs['value'] = options[0] if nonempty else (None, None)
--> 186 super().__init__(*args, **kwargs)
    187 self._initializing_traits_ = False

File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget_description.py:35, in DescriptionWidget.__init__(self, *args, **kwargs)
     33     kwargs.setdefault('tooltip', kwargs['description_tooltip'])
     34     del kwargs['description_tooltip']
---> 35 super().__init__(*args, **kwargs)

File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget.py:503, in Widget.__init__(self, **kwargs)
    501 """Public constructor"""
    502 self._model_id = kwargs.pop('model_id', None)
--> 503 super().__init__(**kwargs)
    505 Widget._call_widget_constructed(self)
    506 self.open()

File ~/openvino_env/lib/python3.10/site-packages/traitlets/traitlets.py:1388, in HasTraits.__init__(self, *args, **kwargs)
   1386 changed = set(kwargs) & set(self._traits)
   1387 for key in changed:
-> 1388     value = self._traits[key]._cross_validate(self, getattr(self, key))
   1389     self.set_trait(key, value)
   1390     changes[key]['new'] = value

File ~/openvino_env/lib/python3.10/site-packages/traitlets/traitlets.py:762, in TraitType._cross_validate(self, obj, value)
    760 if self.name in obj._trait_validators:
    761     proposal = Bunch({"trait": self, "value": value, "owner": obj})
--> 762     value = obj._trait_validators[self.name](obj, proposal)
    763 elif hasattr(obj, "_%s_validate" % self.name):
    764     meth_name = "_%s_validate" % self.name

File ~/openvino_env/lib/python3.10/site-packages/traitlets/traitlets.py:1260, in EventHandler.__call__(self, *args, **kwargs)
   1258 """Pass `*args` and `**kwargs` to the handler's function if it exists."""
   1259 if hasattr(self, "func"):
-> 1260     return self.func(*args, **kwargs)
   1261 else:
   1262     return self._init_call(*args, **kwargs)

File ~/openvino_env/lib/python3.10/site-packages/ipywidgets/widgets/widget_selection.py:244, in _Selection._validate_value(self, proposal)
    242     return findvalue(self._options_values, value, self.equals) if value is not None else None
    243 except ValueError:
--> 244     raise TraitError('Invalid selection: value not found')

TraitError: Invalid selection: value not found`

This error leads to more errors in the following steps such as :`---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[13], line 1
----> 1 text_enc = core.compile_model(TEXT_ENCODER_OV_PATH, device.value)

NameError: name 'device' is not defined
unet_model = core.compile_model(UNET_OV_PATH, device.value)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[14], line 1
----> 1 unet_model = core.compile_model(UNET_OV_PATH, device.value)

NameError: name 'device' is not defined`

I'd really appreciate any help anyone can provide, I'm stumped but would still like to take advantage my Intel arc GPU.

![image_2023-10-18_033620823](https://github.com/openvinotoolkit/openvino_notebooks/assets/125188262/6666277a-fa3c-496e-8e08-f82a86a45424)

![image_2023-10-18_033651014](https://github.com/openvinotoolkit/openvino_notebooks/assets/125188262/2bd64bd0-94c5-4818-817c-4d255cb88fad)

**Comment by brmarkus** [comment]
Do you see any other Jupyter notebook from the collection working - for a consistency-check if your installation is complete?
Do you see notebooks working with the CPU?

From your "About your system" screenshot you have a SoC "intel-core-i5 12600k".
This has an integrated GPU.
From your description you mentioned an "Intel arc GPU".

Assuming your installation is complete and correct - then you might simply need to select the proper GPU device.
As you have at least two Intel GPUs in your system (and maybe even more from other, non-Intel-vendors) - your GPUs can be referenced via different device names, see:
"https://docs.openvino.ai/2023.0/openvino_docs_OV_UG_supported_plugins_GPU.html#device-naming-convention"

The code in your screenshot assumes a single GPU in your system:

```
4 options=core.available_devices + ["AUTO"],
5 value='GPU',
6 description='Device:',
7 disabled=False,
```
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/29976962/aa379392-d056-47be-96b5-de9bf7d52995)

Can you check the value of the variable manually `core.available_devices` - and maybe just change the line `value=`GPU`` to e.g. `value=`GPU.0`` or `value=`GPU.1``? (you would need to check if GPU.0 or GPU.1 is either the integrated or the discrete GPU).
